### PR TITLE
Pushdown Tigris queries with dot notation

### LINKED ISSUE #1841: Pushdown dot notation queries
### What should be done?

We should pushdown filters with keys specified in dot notation format for both pg and tigris handlers

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #1841.

## Readiness checklist

* [x] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/pgdb/query_test.go [hunk 1]
```diff
@@ -281,3 +282,88 @@ func TestGetDocuments(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+// WHERE clauses occurring frequently in test.
+var (
+	whereEq          string = " WHERE ((_jsonb->$1)::jsonb = $2)"
+	whereEqOrContain string = whereEq + " OR (_jsonb->$1)::jsonb @> $2"
```
Comment (chilagrow) [inline internal/handlers/pg/pgdb/query_test.go:289]:
```suggestion
	idWhereClause          string = " WHERE ((_jsonb->$1)::jsonb = $2)"
	fieldWhereClause string =  " WHERE ((_jsonb->$1)::jsonb = $2) OR (_jsonb->$1)::jsonb @> $2"
```
Just naming suggestion.
It's used in one function, doesn't need to be global var. Let's put them in the `TestPrepareWhereClause` function.

File: internal/handlers/pg/pgdb/query_test.go [hunk 2]
```diff
@@ -281,3 +282,88 @@ func TestGetDocuments(t *testing.T) {
...
+)
+
+func TestPrepareWhereClause(t *testing.T) {
+	t.Parallel()
+	objectID := types.ObjectID{0x62, 0x56, 0xc5, 0xba, 0x0b, 0xad, 0xc0, 0xff, 0xee, 0xff, 0xff, 0xff}
+
+	for name, tc := range map[string]struct {
+		filter   *types.Document
+		expected string
+		skip     string
```
Comment (chilagrow) [inline internal/handlers/pg/pgdb/query_test.go:299]: We don't need skip here. Unused :hugs:

File: internal/handlers/tigris/tigrisdb/query_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+func TestBuildFilter(t *testing.T) {
+	t.Parallel()
+	tdb := TigrisDB{}
+	objectID := types.ObjectID{0x62, 0x56, 0xc5, 0xba, 0x0b, 0xad, 0xc0, 0xff, 0xee, 0xff, 0xff, 0xff}
+
+	for name, tc := range map[string]struct {
+		filter   *types.Document
+		expected string
```
Comment (chilagrow) [inline internal/handlers/tigris/tigrisdb/query_test.go:35]:
```suggestion
		expected string // defaults to `{}`
```

**Review by chilagrow — COMMENTED** [review]
Great work!

**Comment by chilagrow** [comment]
Conflict to resolve @noisersup :muscle:

**Review by chilagrow — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]
👍

**Pull request merged**